api.go

```diff
@@ -287,30 +304,24 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 		r.setLeader(r.localAddr)
 	}
 
-	// Restore the current term and the last log
+	// Restore the current term and the last log.
 	r.setCurrentTerm(currentTerm)
 	r.setLastLog(lastLog.Index, lastLog.Term)
 
-	// Attempt to restore a snapshot if there are any
+	// Attempt to restore a snapshot if there are any.
 	if err := r.restoreSnapshot(); err != nil {
 		return nil, err
 	}
 
-	// Scan through the log for any configuration change entries
+	// Scan through the log for any configuration change entries.
 	snapshotIndex, _ := r.getLastSnapshot()
 	for index := snapshotIndex + 1; index <= lastLog.Index; index++ {
 		var entry Log
 		if err := r.logs.GetLog(index, &entry); err != nil {
 			r.logger.Printf("[ERR] raft: Failed to get log at %d: %v", index, err)
 			panic(err)
 		}
-		// TODO: support LogAddPeer, LogRemovePeer too
-		if entry.Type == LogConfiguration {
-			r.configurations.committed = r.configurations.latest
-			r.configurations.committedIndex = r.configurations.latestIndex
-			r.configurations.latest = decodeConfiguration(entry.Data)
-			r.configurations.latestIndex = entry.Index
-		}
+		r.checkAndProcessConfigurationLog(&entry)
```

Comment:
While you're at it, it's not really checking much, so I think you can drop the "checkAnd" too.